cogboard-app/src/main/kotlin/com/cognifide/cogboard/http/HttpClient.kt

```diff
@@ -88,34 +92,63 @@ class HttpClient : AbstractVerticle() {
         val pass = config.getString(CogboardConstants.PROP_PASSWORD) ?: ""
         val token = config.getString(CogboardConstants.PROP_TOKEN) ?: ""
         val headers = config.getJsonObject(CogboardConstants.PROP_HEADERS)
+        val authenticationTypes = Json.decodeValue(config.getString(CogboardConstants.PROP_AUTHENTICATION_TYPES))
+                ?: JsonArray()
 
-        if (user.isNotBlank() && token.isNotBlank()) {
-            request.basicAuthentication(user, token)
-        } else if (user.isNotBlank() && pass.isNotBlank()) {
-            request.basicAuthentication(user, pass)
+        val authenticationType = getAuthenticationType(authenticationTypes as JsonArray, user, token, pass)
+
+        if (authenticationType == AuthenticationType.NONE) {
+            throw Exception("Request cannot be processed. Cause: Incorrect credentials or Widget doesn't supported")
+        } else {
+            AuthenticationFactory(user, pass, token, request).create(authenticationType)
         }
 
         applyRequestHeaders(request, headers)
 
         return request
     }
 
+    private fun getAuthenticationType(authenticationTypes: JsonArray, user: String, token: String, pass: String): AuthenticationType {
```

Comment:
make sure that the order in JsonArray is the same as the initial set by widget

Reply:
i checked and order is correct